app/app.go

```diff
@@ -644,18 +667,29 @@ func New(
 		func(ctx sdk.Context, _ upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
 			ctx.Logger().Info("Handler for upgrade plan: " + UpgradeName)
 
+			// Fix lack of version map initialization in InitChainer for new chains
+			if len(fromVM) == 0 {
```

Comment:
That's what I meant for new vs existing chains.
This is going to fail on upgrading from existing chains since we haven't persisted the introduced params.
e.g. `minCommissionRate`, fee params for did & resource module as well.

So, I'd suggest to enhance this with an `else` block that accounts for existing chains (`len(fromVm) > 0`):

```go
} else {
// Add defaults for did subspace
didSubspace := app.GetSubspace(didtypes.ModuleName)
didSubspace.Set(ctx, didtypes.ParamStoreKeyFeeParams, didtypes.DefaultFeeParams())

// Add defaults for resource subspace
resourceSubspace := app.GetSubspace(resourcetypes.ModuleName)
resourceSubspace.Set(ctx, resourcetypes.ParamStoreKeyFeeParams, resourcetypes.DefaultFeeParams())

// Add defaults for staking subspace
stakingSubspace := app.GetSubspace(stakingtypes.ModuleName)
stakingSubspace.Set(ctx, stakingtypes.KeyMinCommissionRate, sdk.NewDec(0))

// Get version map from previous upgrade
versionMap := app.mm.GetVersionMap()
// Skip capability module
fromVM[capabilitytypes.ModuleName] = versionMap[capabilitytypes.ModuleName]
// Skip distribution module
fromVM[distrtypes.ModuleName] = versionMap[distrtypes.ModuleName]
// Skip gov module
fromVM[govtypes.ModuleName] = versionMap[govtypes.ModuleName]
// Skip mint module
fromVM[minttypes.ModuleName] = versionMap[minttypes.ModuleName]
// Skip staking module
fromVM[stakingtypes.ModuleName] = versionMap[stakingtypes.ModuleName]
// Skip did module (InitGenesis would alter namespace)
fromVM[didtypes.ModuleName] = versionMap[didtypes.ModuleName]
}
```